crates/apollo-compiler/tests/validation/fragments.rs

```diff
@@ -0,0 +1,57 @@
+#[test]
+fn long_fragment_chains_do_not_overflow_stack() {
+    // Build a query that applies thousands of fragments
+    // Validating it would take a lot of recursion and blow the stack
```

Comment:
I take it you’ve manually reproduced the stack overflow when the limit is set too high?

Please also add a similar test that goes right under the limit and succeeds.